docs/provider.md

```diff
@@ -0,0 +1,113 @@
+
+This allows to skip the file name in the sidebar, since it would just be repeated all the time while adding no value to the stream.
+
+The favorite provider makes use of this and the activities don't have additional placeholders, so only the parsed subject is set.
+
+### Long translations with "rich object" string
+
+In the long version for the normal activity stream contains the filename. Objects like users , files and more can be highlighted in the "rich subject", which allows the app to show an avatar next to the name, link the file name to the file list, and many more things.
+
+The `IEvent::setRichSubject()` method has two arguments:
+
+1. The already translated string with placeholders
+2. The list of placeholders, in the case of the favorite provider it's the file that was favorites
+
+```php
+$event->setRichSubject(
+	$this->l->t('You added {file} to your favorites')
+	['file' => [
+		'type' => 'file',
+		'id' => $event->getObjectId(),
+		'name' => basename($event->getObjectName()),
+		'path' => $event->getObjectName(),
+	]]
+);
+```
+
+The list of required keys for each object can be found in the [`OCP\RichObjectStrings\Definitions`](https://github.com/nextcloud/server/blob/01f4c7550538a30311597d4eb9b889fbb04c4d67/lib/public/RichObjectStrings/Definitions.php) class.
+
+### Merging activities
+
+It is also possible to merge activities. E.g. when a user favorites two files, those can be combined to the following:
+
+```php
+$event->setRichSubject(
+	$this->l->t('You added {file1} and {file2} to your favorites')
+	[
+		'file1' => [
+			'type' => 'file',
+			'id' => 23,
+			'name' => 'file1.txt',
+			'path' => 'path1/to1/file1.txt',
+		],
+		'file1' => [
+			'type' => 'file',
+			'id' => 42,
+			'name' => 'file2.txt',
+			'path' => 'path2/to2/file2.txt',
+		],
+	]
+);
+```
+
+Merging two events is fairly easy. Afterwards the child needs to be set, to avoid the previous event from showing up in the stream additionally `$event->setChildEvent($previousEvent)`.
+
+But this should not be done manually! Instead dependency injection should be used to get `OCP\Activity\IEventMerger`. This helper automatically merges events, when the following requirements are met:
+
+* Both events need to have the same `getApp()`
+* Both events must not have a message `getMessage()`
+* Both events need to have the same subject `getSubject()`
+* Both events need to have the same object type `getObjectType()`
+* The time difference between both events must not be bigger then 3 hours
+* Only up to 5 events can be merged.
```

Comment:
why do we have a limit of 5 events? Imagine I upload a series of photos, say a batch of 500. I don't want to see 100 activities.